datanode/integration/integration_test.go

```diff
@@ -100,197 +117,135 @@ func TestMain(m *testing.M) {
-		sortDelegations)
+	goldenDir = filepath.Join(cwd, "testdata", "golden")
+	if err = vgfs.EnsureDir(goldenDir); err != nil {
+		return "", "", fmt.Errorf("couldn't create golden dir: %w", err)
+	}
 
-	assert.Empty(t, diff)
-}
+	vegaHome, err := os.MkdirTemp("", "datanode_test")
+	if err != nil {
+		return "", "", fmt.Errorf("couldn't create temp dir: %w", err)
+	}
 
-func removeDupVotes() cmp.Option {
-	// This is a bit grim; in the old API you get repeated entries for votes when they are updated,
-	// which is a bug not present in the new API - so remove duplicates when comparing (and sort)
-	return cmp.Transformer("DuplicateVotes", func(in []Vote) []Vote {
-		m := make(map[string]Vote)
-		for _, vote := range in {
-			m[fmt.Sprintf("%v-%v", vote.ProposalID, vote.Party.ID)] = vote
-		}
+	postgresRuntimePath := filepath.Join(vegaHome, "pgdb")
 
-		keys := make([]string, len(m))
-		sort.Strings(keys)
+	if err = os.Mkdir(postgresRuntimePath, fs.ModePerm); err != nil {
+		return "", "", fmt.Errorf("couldn't create postgres runtime dir: %w", err)
+	}
 
-		out := make([]Vote, len(m))
-		for i, key := range keys {
-			out[i] = m[key]
-		}
-		return out
-	})
+	return vegaHome, postgresRuntimePath, nil
 }
 
-type queryDetails[T any] struct {
+type queryDetails struct {
 	TestName string
 	Query    string
-	Result   T
+	Result   json.RawMessage
 	Duration time.Duration
 }
 
-func assertGraphQLQueriesReturnSame[T any](t *testing.T, query string) {
+func assertGraphQLQueriesReturnSame(t *testing.T, query string) {
 	t.Helper()
 
 	req := graphql.NewRequest(query)
-	var resp T
-	start := time.Now()
+	var resp map[string]interface{}
+	s := time.Now()
 	err := client.Run(context.Background(), req, &resp)
+	require.NoError(t, err, "failed to run query: '%s'; %s", query, err)
+	elapsed := time.Since(s)
+
+	var respJsn json.RawMessage
+	respJsn, err = json.MarshalIndent(resp, "", "\t")
 	require.NoError(t, err)
-	elapsed := time.Since(start)
 
-	niceName := strings.Replace(t.Name(), "/", "_", -1)
+	niceName := strings.ReplaceAll(t.Name(), "/", "_")
 	goldenFile := filepath.Join(goldenDir, niceName)
 
 	if *writeGolden {
-		details := queryDetails[T]{
+		details := queryDetails{
 			TestName: niceName,
 			Query:    query,
-			Result:   resp,
+			Result:   respJsn,
 			Duration: elapsed,
 		}
-		jsonBytes, err := json.Marshal(details)
+		jsonBytes, err := json.MarshalIndent(details, "", "\t")
 		require.NoError(t, err)
-
-		os.WriteFile(goldenFile, jsonBytes, 0o644)
+		require.NoError(t, os.WriteFile(goldenFile, jsonBytes, 0o644))
 	} else {
 		jsonBytes, err := os.ReadFile(goldenFile)
 		require.NoError(t, err, "No golden file for this test, generate one by running 'go test' with the -golden flag")
-
-		details := queryDetails[T]{}
-		err = json.Unmarshal(jsonBytes, &details)
-		require.NoError(t, err, "Unable to unmarshal golden file")
-
+		details := queryDetails{}
+		require.NoError(t, json.Unmarshal(jsonBytes, &details), "Unable to unmarshal golden file")
```

Comment:
Hey, those are valid points. Basically I couldn't reason with all the structs we were using for deserialisation, also some of them had changed since the integration tests were last used, new ones added and so on... and overall it was just so much complexity in skipping fields and fine-tuning the values comparing them that it would require way too much maintenance. 

`assert.JSONEq` don't care about field order.
And we still have `queryDetails`. The golden files still keep all of those details, just the test result payload is now deserialised into `json.RawMessage`.